internal/build/lifecycle_execution.go

```diff
@@ -706,6 +744,133 @@ func (l *LifecycleExecution) ExtendRun(ctx context.Context, buildCache Cache, ph
 	defer extend.Cleanup()
 	return extend.Run(ctx)
 }
+func (l *LifecycleExecution) ExtendBuildByDaemon(ctx context.Context) error {
```

Comment:
We'll need to write some unit tests for this method, which will involve using a mock for the daemon. This is done in the lifecycle (but mocking the kaniko "dockerfile applier") here: https://github.com/buildpacks/lifecycle/blob/e06b69b5acb14e441659178b13de4a38e7a2e612/extender_test.go#L74 and making assertions here: https://github.com/buildpacks/lifecycle/blob/e06b69b5acb14e441659178b13de4a38e7a2e612/extender_test.go#L233-L256

Reply:
done